src/components/Image/index.tsx

```diff
@@ -31,64 +34,110 @@ function Image({source: propsSource, isAuthTokenRequired = false, session, onLoa
+                }
+                return previousSessionAge.current;
+            }
+            if (isExpiredSession(session.creationDate)) {
+                return new Date().getTime();
+            }
+            return session.creationDate;
+        }
+        return undefined;
+    }, [session, isAuthTokenRequired, isAcceptedSession]);
+    useEffect(() => {
+        if (!isAuthTokenRequired) {
+            return;
+        }
+        previousSessionAge.current = validSessionAge;
+    });
+
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      */
+    // source could be a result of require or a number or an object but all are expected so no unsafe-assignment
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     const source = useMemo(() => {
         if (typeof propsSource === 'object' && 'uri' in propsSource) {
             if (typeof propsSource.uri === 'number') {
                 return propsSource.uri;
             }
             const authToken = session?.encryptedAuthToken ?? null;
             if (isAuthTokenRequired && authToken) {
-                return {
-                    ...propsSource,
-                    headers: {
-                        [CONST.CHAT_ATTACHMENT_TOKEN_KEY]: authToken,
-                    },
-                };
+                if (!!session?.creationDate && !isExpiredSession(session.creationDate)) {
+                    // session valid
+                    return {
+                        ...propsSource,
+                        headers: {
+                            [CONST.CHAT_ATTACHMENT_TOKEN_KEY]: authToken,
+                        },
+                    };
+                }
+                if (session) {
+                    activateReauthenticator(session);
+                }
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                return undefined;
             }
         }
         return propsSource;
         // The session prop is not required, as it causes the image to reload whenever the session changes. For more information, please refer to issue #26034.
+        // but we still need the image to reload sometimes (exemple : when the current session is expired)
+        // by forcing a recalculation of the source (which value could indeed change) through the modification of the variable validSessionAge
         // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
-    }, [propsSource, isAuthTokenRequired]);
+    }, [propsSource, isAuthTokenRequired, validSessionAge]);
 
     /**
      * If the image fails to load and the object position is top, we should hide the image by setting the opacity to 0.
      */
     const shouldOpacityBeZero = isObjectPositionTop && !aspectRatio;
 
-    return (
+    return source === undefined ? (
+        <FullScreenLoadingIndicator />
+    ) : (
```

Comment:
I think if we add our own FullScreenLoadingIndicator here, there will be a gap time between this loading indicator and the LoadingIndicator inside AttachmentModal right? AFAIR, we also have a `FullScreenLoadingIndicator` inside our AttachmentModal.